aws/resource_aws_ebs_default_kms_key.go

```diff
@@ -0,0 +1,80 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsEbsDefaultKmsKey() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsEbsDefaultKmsKeyCreate,
+		Read:   resourceAwsEbsDefaultKmsKeyRead,
+		Update: resourceAwsEbsDefaultKmsKeyUpdate,
+		Delete: resourceAwsEbsDefaultKmsKeyDelete,
+
+		Schema: map[string]*schema.Schema{
+			"key_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateArn,
+			},
+		},
+	}
+}
+
+func resourceAwsEbsDefaultKmsKeyCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	_, err := conn.ModifyEbsDefaultKmsKeyId(&ec2.ModifyEbsDefaultKmsKeyIdInput{
+		KmsKeyId: aws.String(d.Get("key_id").(string)),
+	})
+	if err != nil {
+		return fmt.Errorf("error creating EBS default KMS key: %s", err)
+	}
+
+	d.SetId(resource.UniqueId())
```

Comment:
Should we set this to the key ARN/ID instead of a random identifier so its usable in downstream resources? 😄

Reply:
Currently we are allowing the ARN to change and doing a resource update - If we go with using the ARN as the ID then we need to add `ForceNew: true` to the `key_arn` attribute and remove `resourceAwsEbsDefaultKmsKeyUpdate()`. IMHO this is actually a better way of modeling the resource.